### PR TITLE
Update Segment Identify Event Handler

### PR DESCRIPTION
**Pull request by andriosrobert**
Identify user when email is first available

### CODE REVIEW

**Comment by rogefm** [comment]
Andrios is the opposite, this does not guarantee that the user's data is fresh, the other way did guarantee it, because now you are fetching the user we already have in the state. Before, we first updated the user's state and then proceeded, ensuring the updated user. 
Was there any place where you felt a difference with this implementation?

File: webapp/src/webapp/auth/views/signup.cljs [hunk 1]
```diff
@@ -49,6 +49,7 @@
                            (.preventDefault e)
                            (reset! loading (not @loading))
                            (rf/dispatch [:segment->track "SignUp - Create organization"])
+                           (rf/dispatch [:segment->identify])
```
Comment (rogefm) [inline webapp/src/webapp/auth/views/signup.cljs]: And here you only need to pass the userinfo
```suggestion
                           (rf/dispatch [:segment->identify] (:data @userinfo)
```

**Comment by andriosrobert** [comment]
> Andrios is the opposite, this does not guarantee that the user's data is fresh, the other way did guarantee it, because now you are fetching the user we already have in the state. Before, we first updated the user's state and then proceeded, ensuring the updated user. Was there any place where you felt a difference with this implementation?

@rogefm this was a bad description, my main goal here was to trigger the identify call, the potential improvement is secondary.

**Comment by andriosrobert** [comment]
And from what I understood the previous identify implementation didn't use the user information, it had only the user id pulled from the user received as a parameter

also: I didn't find calls to the identify method, I don't think it was being used before

**Comment by rogefm** [comment]
> And from what I understood the previous identify implementation didn't use the user information, it had only the user id pulled from the user received as a parameter
> 
> also: I didn't find calls to the identify method, I don't think it was being used before

Yes they do! here:
`(rf/reg-event-fx
 :segment->identify
 (fn [{:keys [db]} [_ user]]
   (let [user-id (:id user)
         analytics (-> db :segment->analytics)]
     (if (nil? analytics)
       {:fx [[:dispatch [:segment->load [:segment->identify user]]]]}
       (do
         (.identify analytics user-id (clj->js user))   <-------
         {})))))`

**Comment by rogefm** [comment]
> > Andrios is the opposite, this does not guarantee that the user's data is fresh, the other way did guarantee it, because now you are fetching the user we already have in the state. Before, we first updated the user's state and then proceeded, ensuring the updated user. Was there any place where you felt a difference with this implementation?
> 
> @rogefm this was a bad description, my main goal here was to trigger the identify call, the potential improvement is secondary.

Cool, so can we keep it the same as before? The improvement can provide false data if the user is updated any time ago.

File: webapp/src/webapp/auth/views/signup.cljs [hunk 2]
```diff
@@ -49,6 +49,7 @@
                            (.preventDefault e)
                            (reset! loading (not @loading))
                            (rf/dispatch [:segment->track "SignUp - Create organization"])
+                           (rf/dispatch [:segment->identify] (:data @userinfo)
```
Comment (andriosrobert) [inline webapp/src/webapp/auth/views/signup.cljs:52]: @rogefm do we have userinfo at this point? note that it is right before creating the org. the main thing I'm looking for here is the email to be present

**Review by andriosrobert — COMMENTED** [review]
I rolled back the identify change

**Review by rogefm — APPROVED** [review]

**Pull request merged**